src/components/Nav/index.tsx

```diff
@@ -92,16 +104,44 @@ const Nav: FC<IProps> = ({ path }) => {
                   variant="ghost"
                   isSecondary
                   px={1.5}
+                  _hover={{
+                    transform: "rotate(30deg)", // Rotate the icon to 30 degrees on hover
+                    color: "blue", // Change color to blue on hover
+                  }}
                   onClick={toggleColorMode}
                 ></IconButton>
+
                 <ButtonLink
+                  onMouseOver={() => setLanguagesHover(true)}
+                  onMouseOut={() => setLanguagesHover(false)}
+                  as={Link}
                   to={`/languages/${fromPageParameter}`}
-                  leftIcon={<Icon as={MdLanguage} />}
                   variant="ghost"
                   isSecondary
                   px={1.5}
+                  _hover={{
+                    color: "blue", // Change color to blue on hover
+                  }}
+                  style={{
+                    display: "inline-flex",
+                    alignItems: "center",
+                  }}
                 >
-                  {t("languages")} {i18n.language.toUpperCase()}
+                  <Box mr={2} mt={2}>
+                    {" "}
+                    {/* Add spacing between the text and the icon */}
+                    <Icon
+                      as={MdLanguage}
+                      style={{
+                        transition: "transform 0.3s ease-in-out, color 0.2s", // Apply the transition to the icon
+                        transform: languagesHover
+                          ? "rotate(30deg)"
+                          : "rotate(0deg)", // Rotate the icon to 30 degrees on hover
```

Comment:
There are a couple of pretty cool things to do here. Both `transition` and `transform` are style props, so they should not be under the `style` prop.

Secondly, we can put `transform` under the hover pseudo and not waste energy on `useState`. This is a two step process. 
1. Add `data-group` to `ButtonLink`
2. In the `Icon` use the `_groupHover` prop which will target `data-group` with a descendant selector. so when hovering `ButtonLink` the styles under this prop will trigger. Then no need for this ternary!

So in short...
```tsx
<ButtonLink
  // -- Other props -- //
  data-group
  leftIcon={
    <Icon
      as={MdLanguage}
      transition="transform 0.3s ease-in-out, color 0.2s"
      _groupHover={{
        transform: "rotate(30deg)"
      }}
    />
  }
/>
```

Reply:
@rohan9024 marking this as unresolved.

Currently we have the `_groupHover` but not the `data-group` set in the parent ButtonLink.